orchestrator/translator.js

```diff
@@ -51,87 +65,151 @@ function addNegatedFixedEventCondition(node, request) {
           ruleType = node.rules[ruleIx].v2t;
           addFixedEventCondition(node, 'gte', ruleValue, ruleType, request);
           break;
-        case 'else':
-          // This is this node!
-          break;
         default:
           addFixedEventCondition(node, orchtypes.NodeRed.NegatedLogicalOperators[ruleOperation], ruleValue, ruleType, request);
       }
+    } else {
+      if (ruleOperation in orchtypes.NodeRed.LogicalOperators) {
+        ret.status = 'operator has no negated form';
+      } else {
+        ret.code = -1;
+        ret.status = 'invalid operator'
+      }
     }
   }
+
+  return ret;
 }
 
 function addEventCondition(node, ruleOperation, ruleValue, ruleType, request, eventConditionArray) {
+  let ret = {
+    'status' : 'ok',
+    'code': 0
+  }
+  // Sanity checks
+  for (let i = 0; i < arguments.length; i++) {
+    if (arguments[i] == undefined || arguments[i] === null) {
+      ret.status = 'invalid parameter';
+      ret.code = -1;
+      return ret;
+    }
+  }
+  // End of sanity checks
+
   if (ruleOperation in orchtypes.NodeRed.LogicalOperators) {
     let nodeProperty = tools.trimProperty(node.property, '.');
     tools.addUniqueToArray(request.variables, nodeProperty);
     tools.addUniqueToArray(request.inputDevice.attributes, nodeProperty);
     eventConditionArray.push({'q' : nodeProperty + ' ' + orchtypes.NodeRed.LogicalOperators[ruleOperation] + ' ' + ruleValue });
   } else if (isValidGeoOperator(ruleOperation)) {
     // Sanity checks
-    if (ruleType == undefined || ruleValue == undefined || (ruleValue != undefined && ruleValue.length == 0)) {
-      throw {retCode: 400, msg:'empty georeference node'}
-    }
-
-    if (ruleType == orchtypes.NodeRed.GeoFenceMode.POLYLINE) {
-      // For now, georeferenced tests uses only one attribute.
-      let expression = {
-        'georel': ruleOperation,
-        'coords': '',
-        'geometry': ''
-      };
-      expression.geometry = orchtypes.OrionTypes.GeoFenceMode.POLYGON;
-      expression.coords = '';
-      for (let i = 0; i < ruleValue.length; i++) {
-        let point = ruleValue[i];
-        expression.coords += point.latitude + ',' + point.longitude + ';';
+    if (ruleValue.length == 0) {
+      ret.status = 'empty georeference node';
```

Comment:
can we add the id of the invalid node in the error object? (that way we might help the user identify the invalid node)

Reply:
Sure!